torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -239,6 +239,9 @@ const IValue& OwnerRRef::getValue() const {
   if (future_->hasError()) {
     (void)future_->value(); // Throws the error.
   }
+  // Before accessing the value in this RRef, current CUDA streams must wait
+  // for pending CUDA operations that create the value.
+  RRefContext::getInstance().blockCurrentStreams(events_);
```

Comment:
I know nothing about RRefs, hence let me ask: a few lines above we call `future_->wait()`, what future is that? What value does it hold? If the future is a CUDAFuture which holds the value, doesn't it already store some events on its own, and wouldn't calling `wait()` already synchronize those events?

Reply:
Currently it's an `ivalue::Future`. I do plan to convert that to a CUDAFuture on-demand when  we have a consensus on how to record devices in non-RPC RRefs

Reply:
Let me reiterate that I know nothing about RRefs, but, if we changed the future to a CUDAFuture now could that help solve any of our problems? (current, or even past problems)

I'd find it cleaner to have all the synchronization logic in a single place (CUDAFuture) and have everything else just leverage that logic instead of reimplementing it.